app/Controllers/Http/ProfileController.ts

```diff
@@ -0,0 +1,137 @@
+import InvalidDataError from "App/Exceptions/CustomErrors/InvalidDataError";
+import ResourceExistError from "App/Exceptions/CustomErrors/ResourceExistError";
+import UnauthorizedError from "App/Exceptions/CustomErrors/UnauthorizedError";
+import UndefinedFieldError from "App/Exceptions/CustomErrors/UndefiendFieldError";
+import Profile from "App/Models/Profile";
+import User from "App/Models/User";
+import UserProfileValidator from "App/Validators/UserProfileValidator";
+import moment from "moment";
+
+export default class ProfileController {
+
+  //For creating the registered user profile
+  public async createuserProfile({ request, response, auth }) {
+    try {
+      const loginUserId = auth.user.id;
+      const profile = await Profile.findBy("user_id", loginUserId);
+
+      //Checking if the profile already exist
+      if(profile !== null){
+        throw new ResourceExistError("Profile already exists");
+      }else{
+        const { name, gender, mobileNumber, dateOfBirth } =
+          await request.validate(UserProfileValidator);
+        //Getting the age by using moment package
+        const age = moment().diff(new Date(dateOfBirth), "years");
+        //Finding the user based on mobile number, for checking same mobile number doesn't belongs to other user
+        const existedProfile = (mobileNumber !== undefined) && await Profile.findBy("mobile_number", mobileNumber);
```

Comment:
mobile number will be a required field then why are we checking for undefined value here ?